sp-files/.github/workflows/markdown-style-checks.yml

```diff
@@ -0,0 +1,15 @@
+name: "Linter for Markdown"
+
+on: [push, pull_request]
```

Comment:
We probably can limit it to avoid unnecessary CI runs

However, if you want that the teams never touch this file, then keep your implementation. 

```
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - '*'
```